### PR TITLE
filter and search

### PR DESCRIPTION
**Pull request by prakashsakari**
This PR has the following
1. filter by category
2. filter by most played
3. search quiz by title

Link: https://deploy-preview-16--groovyquiz.netlify.app/

### CODE REVIEW

File: src/components/Filter/MostPlayed.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { useFilter } from "../../context";
+
+export const MostPlayed = () => {
+  const {
+    filterState: { isMostPlayed },
+    filterDispatch
+  } = useFilter();
+
+  console.log(isMostPlayed);
```
Comment (ranitasahaa1230) [inline src/components/Filter/MostPlayed.jsx:9]: Remove this console.log if not needed

**Review by ranitasahaa1230 — APPROVED** [review]
Good use of contexts and reducers
Code can be understood easily

File: src/components/Filter/Category.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      <div className="select-girls margin8-top-down">
+        <label className="girls d-flex gap">
+          <input
+            className="check-box"
+            type="checkbox"
+            value="vehicles"
+            checked={category === "vehicles"}
+            onChange={(e) => handleChange(e)}
+          />
+          Vehicles
```
Comment (ivinayakg) [inline src/components/Filter/Category.jsx:91]: make the options dynamic for component scalability

File: src/db/categories.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import geog from "../assets/geog.svg";
+import history from "../assets/history.svg";
+import math from "../assets/maths.svg";
+import book from "../assets/book.svg";
+import comics from "../assets/comics.svg";
+import science from "../assets/science.svg";
+import tech from "../assets/tech.svg";
+import sports from "../assets/sports.svg";
+import vehicle from "../assets/vehicle.svg";
```
Comment (ivinayakg) [inline src/db/categories.js:9]: add the asset to the public folder and then access it

**Review by ivinayakg — APPROVED** [review]
good work 👍

File: src/components/Auth/Login.jsx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import "./Auth.css";
 import { Link } from "react-router-dom";
-import { useAuth } from "../../context/auth-context";
+import { useAuth } from "../../context";
```
Comment (nayyyhaa) [inline src/components/Auth/Login.jsx:3]: Suggestion: you can use absolute import for better code readability.  Read: https://dev.to/emanuelone/how-to-use-absolute-path-in-reactjs-503i

**Review by nayyyhaa — APPROVED** [review]
Nice work with the UI + code is well structured. 💯 

One suggestion, I noticed that I was able to select only one checkbox at a time, you can use radio-button for that purpose OR allow the user to select more than one option using checkbox.

image

File: src/components/Filter/MostPlayed.jsx (hunk 1 above)
Comment (prakashsakari) [reply src/components/Filter/MostPlayed.jsx:9]: thanks, updated

File: src/components/Filter/Category.jsx (hunk 2 above)
Comment (prakashsakari) [reply src/components/Filter/Category.jsx:91]: ok will do that after submission

File: src/db/categories.js (hunk 3 above)
Comment (prakashsakari) [reply src/db/categories.js:9]: ok will do that

File: src/components/Auth/Login.jsx (hunk 4 above)
Comment (prakashsakari) [reply src/components/Auth/Login.jsx:3]: will read about this, thanks

**Pull request merged**